tests/integration/tier1/single-yum-transaction/test_check_for_latest_packages.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+
+from conftest import SYSTEM_RELEASE_ENV
+from envparse import env
+
+
+@pytest.mark.test_packages_upgraded_after_conversion
+def test_packages_upgraded_after_conversion(convert2rhel, shell):
+    """
+    Verify that packages get correctly reinstalled and not
+    downgraded during the conversion.
+    """
+
+    checked_packages = ["shim-x64"]
+
+    if "oracle-7" in SYSTEM_RELEASE_ENV:
+        checked_packages = [
+            "krb5-libs.x86_64",
+            "nss-softokn-freebl.x86_64",
+            "nss-softokn.x86_64",
+            "expat.x86_64",
+            "krb5-libs.x86_64",
+        ]
+
+    packages_to_verify = {}
+
+    for package in checked_packages:
+        latest_version = shell(f"repoquery --quiet --latest-limit=1 {package}").output.strip("\n")
+        is_installed = shell(f"rpm -q {package}").output
+        if "is not installed" in is_installed:
+            shell(f"yum install -y {package}")
+        # although not really used, keep the assembled dictionary if needed for version comparison
+        # also the dictionary gets appended only when there is the package available to install
+        # (repoquery yields a value)
+        if latest_version:
+            packages_to_verify[package] = latest_version
+
+    # Run utility until the reboot
+    with convert2rhel(
+        "-y --no-rpm-va --serverurl {} --username {} --password {} --pool {} --debug".format(
+            env.str("RHSM_SERVER_URL"),
+            env.str("RHSM_USERNAME"),
+            env.str("RHSM_PASSWORD"),
+            env.str("RHSM_POOL"),
+        )
+    ) as c2r:
+        c2r.expect("Conversion successful!")
+    assert c2r.exitstatus == 0
+
+    for package in packages_to_verify:
+        # If tha package lands on latest version after conversion
+        # `yum check-update` will return 0
+        # If the package is possible to update the returncode yields 100
```

Comment:
```suggestion
        # If it is possible to update the package, the yum returncode yields 100
```

Reply:
Will update in separate PR to unblock merge